src/rcc.rs

```diff
@@ -1,26 +1,341 @@
 //! Reset and Clock Control
 
-use stm32l4x6::{rcc, RCC};
+// TODO right now the various configure functions reach into rcc directly. This is bad. Add an
+// opaque CR member to RCC, and add methods to CR and BDCR. They should probably take clock source
+// variant arguments.
 
-use ::cmp;
+use stm32l4x6::{rcc, RCC, PWR};
 
 use common::Constrain;
 use time::Hertz;
 use flash::ACR;
 
 impl Constrain<Rcc> for RCC {
+    /// Create an RCC peripheral handle.
+    ///
+    /// Per 6.2, "The MSI is used as system clock source after startup from Reset, configured
+    /// at 4 MHz."
     fn constrain(self) -> Rcc {
+        // Enable write access to the BDCR
+        unsafe { (*PWR::ptr()).cr1.modify(|_,w| w.dbp().set_bit()); }
```

Comment:
afaik isn't BDCR is not necessary to be written.
Or do you think it is ok to lift write protection even if we are not going to write there?

Please add the comment about it to documentation on RCC struct